### PR TITLE
Update DG with usecases

### PR DESCRIPTION
**Pull request by lordidiot**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -650,10 +635,10 @@ Use case ends.
 * 1a. System detects an error in the entered data.
     * 1a1. System shows an error message.
     * 1a2. User enters a new add request. <br>
-      Steps 1a1- 1a2 are repeated until the data entered are correct.<br>
+      Steps 1a1- 1a2 are repeated until the data entered is correct.<br>
       Use case resumes from step 2. <br>
 
-**Use case: UC3 - List all contacts**
+**Use case: UC2 - List all contacts**
```
Comment (limjunxian1) [inline docs/DeveloperGuide.md:641]: under this use case, just realized that step 2 is `System adds the new contact into its system`. should change to `System displays all the contacts saved in its system`.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -623,23 +623,8 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-
-**Extensions**
-   * 1a. System detects an error in data entered.
-     * 1a1. System shows the correct format for request.
-     * 1a2. User enters a new opt out request. <br>
-     Steps 1a1-1a2 are repeated until the data entered are correct. <br>
-     Use case resumes from step 4. <br>
-
-**Use case: UC2 - Add new contact**
+**Use case: UC1 - Add new contact**
```
Comment (limjunxian1) [inline docs/DeveloperGuide.md:627]: this one the second step of the use case is `System shows a list of all contacts.` should change to `System adds the new contact into its system.`

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -718,29 +703,59 @@ Use case ends.
...
-  Steps 2a1- 2a2 are repeated until the data entered are correct.<br>
+  Steps 2a1- 2a2 are repeated until the data entered is correct.<br>
   Use case resumes from step 3. <br>
 
-**Use Case UC7: Add emergency contact**
+**Use Case UC6 - Add note**
+
+**MSS**
+1. User <ins>lists all contacts (UC2).</ins>
+2. User enters an index to add a note associated with a contact
```
Comment (limjunxian1) [inline docs/DeveloperGuide.md]: here, this sounds like user just enters index for this command.

```suggestion
2. User enters an index of the specified contact alongside with the notes to be added.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -718,29 +703,59 @@ Use case ends.
...
+3. System adds the note to the contact.<br>
+Use case ends.
+
+**Extensions**
+* 2a. System detects an invalid index entered.
+  * 2a1. System shows an error message.
+  * 2a2. User tries to add a note to a contact again.<br>
```
Comment (limjunxian1) [inline docs/DeveloperGuide.md]: should we standardize this with the above UCs?

```suggestion
  * 2a2. User enters a new addnote request.
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -718,29 +703,59 @@ Use case ends.
...
+
+
+**Use Case UC7 - Record payment**
 
 **MSS**
-1. User <ins>lists all contacts (UC3).</ins>
-2. User adds a contact to the emergency contact list.
-3. System adds the contact into the emergency contact list. <br>
+1. User <ins>lists all contacts (UC2).</ins>
+2. User enters an index to record an amount of money paid to a contact.
```
Comment (limjunxian1) [inline docs/DeveloperGuide.md]: afaik, it is not necessarily `paid` but also, there is a concept of `owe`. perhaps we can combine them into this single use case?

```suggestion
2. User enters an index of the specified contact alongside with the amount of money paid/owed to that contact.
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -718,29 +703,59 @@ Use case ends.
...
+3. System records money paid to the contact.
```
Comment (limjunxian1) [inline docs/DeveloperGuide.md]: to change to `paid/owed` if decide to combine into a single use case

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -718,29 +703,59 @@ Use case ends.
...
 Use case ends.
 
 **Extensions**
-* 1a. System detects that the tag given to the contact is invalid.
-  * 1a1. System shows the valid tags for input.
-  * 1a2. User enters a new add emergency contact request. <br>
-  Steps 1a1- 1a2 are repeated until the data entered are correct. <br>
+* 2a. System detects an invalid index entered.
+  * 2a1. System shows an error message.
+  * 2a2. User tries to record a payment to a contact again.<br>
```
Comment (limjunxian1) [inline docs/DeveloperGuide.md]: ditto, see above comments for standardization

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -718,29 +703,59 @@ Use case ends.
...
+
+**MSS**
+1. User enters a search criteria to find a contact.
+2. System displays all contacts that match the criteria.<br>
+Use case ends.
+
+**Extensions**
+* 1a. System detects an invalid search criteria.
+  * 1a1. System shows an error message.
+  * 1a2. User tries to enter a search criteria to find a contact.<br>
```
Comment (limjunxian1) [inline docs/DeveloperGuide.md]: ditto, see above comments

**Review by limjunxian1 — COMMENTED** [review]
Overall, LGTM! thanks for tidying up the use cases but there are some use cases currently missing also, e.g. `updatephoto` and `removenote`. otherwise, happy to merge this!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (lordidiot) [reply docs/DeveloperGuide.md:641]: Done!

**Comment by lordidiot** [comment]
Thank you for the feedback @limjunxian1, have addressed the issues. PTAL!

**Comment by limjunxian1** [comment]
thanks @lordidiot for the quick fixes! LGTM and thank you for writing use cases on the group's behalf!

**Review by limjunxian1 — APPROVED** [review]

**Pull request merged**